orchestrator/services/processes.py

```diff
@@ -166,6 +167,8 @@ def _db_log_step(stat: ProcessStat, step: Step, process_state: WFProcess) -> WFP
         db.session.rollback()
         raise
 
+    send_data_to_websocket(stat.pid, step, step_state, process_state.status)
```

Comment:
Maybe this is is bit too tightly coupled. A full-blown event mechanism might be a bit over the top now, but we have to think of some hooks that you can use instead of direct calls. For a first implementation this might be ok.

Reply:
We will probably resolve this whilst implementing the other tickets. Lets leave it for now :)